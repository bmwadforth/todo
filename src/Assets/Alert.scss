@import "Variables";

.alert {
  width: 100%;
  .alert-body {
    height: 250px;
    &.alert-success {
      background-color: $success;
    }

    &.alert-danger {
      background-color: $danger;
    }
  }
}
