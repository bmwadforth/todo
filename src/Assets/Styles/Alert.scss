@import "Variables";

.alert {
  width: 400px;
  margin: 1em 0;
  .alert-body {
    padding: 1em;
    border-radius: $border-radius;
    min-height: 50px;
    height: auto;
    &.alert-primary {
      background-color: $primary;
      svg {
        path {
          fill: #FFF !important;
        }
      }
      color: #FFF;
    }

    &.alert-success {
      background-color: $success;
    }

    &.alert-danger {
      background-color: $danger;
    }

    .alert-title {
      display: flex;
      flex-direction: row;
      align-items: center;
      .icon {
        svg {
          height: 25px;
        }

      }

      p {
        margin: 0;
      }

    }

    .alert-content {

    }
  }
}
