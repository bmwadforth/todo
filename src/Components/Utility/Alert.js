import React from "react";
import PropTypes from "prop-types";
import Icon, {ICONS} from "./Icons";

export default function Alert(props){
    function loadIcon(){
        switch(props.status){
            case "success":
                return ICONS.SUCCESS;
            case "danger":
                return ICONS.ERROR;
            default:
                return ICONS.ERROR;
        }
    }

    return (
        <div className="alert">
            <div className={`alert-body alert-${props.status}`}>
                <div className="alert-title">
                    <Icon icon={loadIcon()}/>
                    <p>{props.title}</p>
                </div>
                <div className="alert-content">
                    {props.children}
                </div>
            </div>
        </div>
    )
}

Alert.propTypes = {
    title: PropTypes.string.isRequired,
    status: PropTypes.oneOf(["success", "danger"]).isRequired
};
