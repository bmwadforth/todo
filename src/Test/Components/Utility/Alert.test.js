import React from 'react';
import Alert from "../../../Components/Utility/Alert";
import {shallow, configure} from "enzyme";
import Adapter from 'enzyme-adapter-react-16';
configure({adapter: new Adapter()});

describe("Alert Component", () => {
    it("Should Match Snapshot", () => {
        const wrapper = shallow(<Alert title="Test" status="success"/>);
        expect(wrapper.html()).toMatchSnapshot();
    });

    it("Should Render With Success Class", () => {
        const wrapper = shallow(<Alert title="Test" status="success"/>);
        expect(wrapper.find(".alert-success")).toBeTruthy();
    });
});
