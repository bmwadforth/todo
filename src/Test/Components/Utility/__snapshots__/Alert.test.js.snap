// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alert Component Should Match Snapshot 1`] = `"<div class=\\"alert\\"><div class=\\"alert-body alert-success\\"><div class=\\"alert-title\\"><div class=\\"icon\\"><span><svg>ok.svg</svg></span></div><p>Test</p></div><div class=\\"alert-content\\"></div></div></div>"`;
